### PR TITLE
Adding video player to editor page - AB#16699

### PR DESCRIPTION
**Pull request by tmiller-msft**
Initial implementation of simple video player on editor page
- Includes video player with customizable playback speed
- Auto plays selected videos
- No longer auto plays videos in side bar
- Supports custom image to display when video not playing (defaults to 1st video frame)
- Updates tests
- Refactors a few other tests to centralize test asset creation
- Updates version of React to 16.7.0
- Rename IVideoSettings to IProjectVideoSettings for clarity

### CODE REVIEW

**Comment by tmiller-msft** [comment]
>           Looks good so far! We should sync up though since we want to get this to interface with the canvastools component on the main editor page. There are some questions surrounding rendering/tagging that we can go over to make sure it will all play nice

It's not often someone is awake and updating PRs the same time I am! Let's chat in morning.

**Comment by kichinsky** [comment]
>           Looks good so far! We should sync up though since we want to get this to interface with the canvastools component on the main editor page. There are some questions surrounding rendering/tagging that we can go over to make sure it will all play nice

Yeah, `canvastools` renders video frame on the `canvas` element, using a `video`-element as input (usually an invisible one), but it does not manage any video control elements as it is out of the tagging scope.

**Comment by tmiller-msft** [comment]
> > ```
> >       Looks good so far! We should sync up though since we want to get this to interface with the canvastools component on the main editor page. There are some questions surrounding rendering/tagging that we can go over to make sure it will all play nice
> > ```
> 
> Yeah, `canvastools` renders video frame on the `canvas` element, using a `video`-element as input (usually an invisible one), but it does not manage any video control elements as it is out of the tagging scope.

Yeah, shouldn't be a problem, we can get the video element for input here as well.

File: src/common/mockFactory.ts [hunk 1]
```diff
@@ -53,14 +53,40 @@ export default class MockFactory {
...
         const assets: IAsset[] = [];
-        for (let i = 1; i <= count; i++) {
-            assets.push(MockFactory.createTestAsset(i.toString()));
+        for (let i = 1; i <= count / 2; i++) {
+            assets.push(MockFactory.createVideoTestAsset(i.toString()));
+        }
+        if (count > 1) {
+            for (let i = 1; i <= count / 2; i++) {
+                assets.push(MockFactory.createTestAsset(i.toString()));
+            }
```
Comment (mydiemho) [inline src/common/mockFactory.ts]: why the use of 2 for loop and a conditional? Is there any way we could simplify this?

File: src/react/components/pages/editorPage/editorSideBar.test.tsx [hunk 2]
```diff
@@ -13,7 +14,7 @@ describe("Editor SideBar", () => {
 
     it("Component renders correctly", () => {
         const props: IEditorSideBarProps = {
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/editorSideBar.test.tsx]: I see that props is the same in many places, could we extract it to the before section?

File: src/react/components/pages/editorPage/editorSideBar.test.tsx [hunk 3]
```diff
@@ -84,21 +85,37 @@ describe("Editor SideBar", () => {
...
-            type: AssetType.Image,
+    it("Correctly switches from image to video and back", (done) => {
+        const testAssets = MockFactory.createTestAssets();
+        const props: IEditorSideBarProps = {
+            assets: testAssets,
+            selectedAsset: testAssets[0],
+            onAssetSelected: onSelectAssetHanlder,
+        };
+
+        const nextAsset = testAssets[6];
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/editorSideBar.test.tsx]: why 6? seems magical

File: src/react/components/pages/editorPage/editorSideBar.tsx [hunk 4]
```diff
@@ -90,13 +90,18 @@ export default class EditorSideBar extends React.Component<IEditorSideBarProps,
...
+            shouldShowPlayControls: false,
+        };
 
         return (
             <div key={key} style={style}
                 className={this.getAssetCssClassNames(asset, selectedAsset)}
                 onClick={() => this.onAssetClicked(asset)}>
                 <div className="asset-item-image">
-                    <AssetPreview asset={asset} />
+                    <AssetPreview asset={asset} videoSettings={editorVideoSetting} />
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/editorSideBar.tsx]: Nit: `editoVideoSettings` to match `videoSettings`

File: src/react/components/pages/editorPage/editorSideBar.test.tsx [hunk 5]
```diff
@@ -84,21 +85,37 @@ describe("Editor SideBar", () => {
...
+        const wrapper = createComponent(props);
+        wrapper.setProps({
+            selectedAsset: nextAsset,
+        });
+
+        setImmediate(() => {
+            expect(wrapper.state()["selectedAsset"]).toEqual(nextAsset);
```
[the remark this answers is not in the document]
Comment (tmiller-msft) [reply src/react/components/pages/editorPage/editorSideBar.test.tsx]: yeah, this is just copied from the previous test, and I didn't want to go back and update all of the tests, so choose to be consistent instead. :)

File: src/react/components/pages/editorPage/editorPage.scss [hunk 6]
```diff
@@ -1,4 +1,5 @@
 @import '../../../../assets/sass/theme.scss';
+@import "~video-react/styles/scss/video-react";
```
[the remark this answers is not in the document]
Comment (tmiller-msft) [reply src/react/components/pages/editorPage/editorPage.scss]: Should be able to override everything.

File: src/common/mockFactory.ts (hunk 1 above)
Comment (tmiller-msft) [reply src/common/mockFactory.ts]: ie, I could have done this all in one loop, such as:
```javascript
for (let i = 1; i <= count; i++) {
   assets.push((i % 2 == 0) ? MockFctory.createTestAsset(i.toString()) | MockFactory.createVideoTestAsset(i.toString())));
}
```
But I think the explicit version is more clear.

**Pull request merged**